packages/destination-actions/src/destinations/fullstory/request-params.ts

```diff
@@ -160,3 +159,66 @@ export const createUserRequestParams = (settings: Settings, requestBody: Object)
     }
   }
 }
+
+/**
+ * Returns {@link RequestParams} for the V2 Create Events HTTP API endpoint.
+ *
+ * @param settings Settings configured for the cloud mode destination.
+ * @param requestValues Values to send with the request.
+ */
+export const createEventRequestParams = (
+  settings: Settings,
+  requestValues: {
+    userId: string
+    eventName: string
+    properties: {}
+    timestamp?: string
+    useRecentSession?: boolean
+    sessionUrl?: string
+  }
+): RequestParams => {
+  const { userId, eventName, properties: eventData, timestamp, useRecentSession, sessionUrl } = requestValues
+  const defaultParams = defaultRequestParams(settings, `v2beta/events?${integrationSourceQueryParam}`)
+
+  const requestBody: Record<string, any> = {
+    name: eventName,
+    properties: eventData
+  }
+
+  if (userId) {
+    requestBody.user = {
+      uid: userId
+    }
+  }
+
+  if (timestamp) {
+    requestBody.timestamp = timestamp
+  }
+
+  if (sessionUrl) {
+    requestBody.session = {
+      id: decodeURIComponent(sessionUrl.substring(sessionUrl.lastIndexOf('/') + 1))
+    }
+  }
+  if (useRecentSession) {
+    // TODO(mattgrosvenor): We're intentionally omitting the use_recent_session request param when the given value is false. At
+    // time of writing, the API will treat use_recent_session=false as use_recent_session=true. This can be removed in
+    // the future when this bug is fixed in the API.
```

Comment:
Do we know if this is still an issue we need to work around in V2?

Reply:
Oof, I wasn't aware of this. I'm not seeing a Bug in JIRA. @mattgrosvenor-fs do you know if one already exists?

Reply:
I'm not aware of a bug in JIRA. This has always existed as far as I know in both V1 and V2.

Bug created in JIRA.